lib/exandra.ex

```diff
@@ -327,6 +327,39 @@ defmodule Exandra do
 
     {keyspace, conn}
   end
+
+  @doc """
+  Streams the results of a simple query or a prepared query to the given callback.
+  The argument will be a `Xandra.PageStream` that you can iterate on to process
+  each `Xandra.Page and accumulate any results you need.
+
+  See `Xandra.prepare!/4` and `Xandra.stream_pages!/4` for more information including
+  supported options.
+
+  ## Examples
+
+      Exandra.stream!(
+        {"SELECT * FROM USERS WHERE can_contact = ?", [true]},
+        MyRepo,
+        fn page ->
+          Enum.each(stream, fn page ->
+            Enum.each(page, &send_email(&1.email, "Hello!")
+          end)
+        end
+      )
+  """
+  @spec stream!({String.t(), list(term())}, atom(), function(), Keyword.t()) :: term()
```

Comment:
I think here we don't need to have a tuple with SQL + params, right? Something like

```suggestion
  @spec stream!(Ecto.Repo.t(), sql :: String.t(), values :: [term()], function(), Keyword.t()) :: term()
```

Reply:
We can yes. This made it compatible with doing something directly with an Ecto query using `Repo.to_sql/2` which returns `{String.t(), []}`. Is it preferable to just accept the sql and the params separately, or have both?

Examples:
```elixir
:all
|> Repo.to_sql(from thing in MyThing, where: thing.name == "test")
|> Exandra.stream(Repo, &Enum.to_list/1)

{sql, params] = Repo.to_sql(:all, from(thing in MyThing, where: thing.name == "test"))
Exandra.stream(sql, params, Repo, &Enum.to_list/1)
```